includes/class-wc-payments-order-success-page.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Class WC_Payments_Order_Service
+ *
+ * @package WooCommerce\Payments
+ */
+
+/**
+ * Class handling order success page.
+ */
+class WC_Payments_Order_Success_Page {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_filter( 'woocommerce_thankyou_order_received_text', [ $this, 'show_woopay_thankyou_notice' ], 10, 2 );
+		add_action( 'woocommerce_thankyou_woocommerce_payments', [ $this, 'show_woopay_payment_method_name' ] );
+		add_action( 'wp_enqueue_scripts', [ $this, 'enqueue_scripts' ] );
+	}
+
+
+	/**
+	 * Return the WooPay thank you notice when the order was created via WooPay
+	 *
+	 * @param string   $text  the default thank you text.
+	 * @param WC_Order $order the order being shown.
+	 *
+	 * @return string
+	 */
+	public function show_woopay_thankyou_notice( $text, $order ) {
+		if ( ! $order->get_meta( 'is_woopay' ) ) {
+			return $text;
+		}
+
+		return '<div class="thankyou-notice-woopay">' . __( 'Thank you! We’ve received your order.', 'woocommerce-payments' ) . '</div>';
+	}
+
+
+	/**
+	 * Add a WooPay logo and card last 4 to the payment method name in the success page.
+	 *
+	 * @param int $order_id the order id.
+	 */
+	public function show_woopay_payment_method_name( $order_id ) {
+		$order = new WC_Order( $order_id );
```

Comment:
Here too, I think it's better to use `wc_get_order` and then add a check for if it returns `false` in the conditional below.

Reply:
Fixed with bfa9b7a5fea1faed1f54312ff1bcd004032a09db